common/schedule.go

```diff
@@ -224,36 +225,30 @@ func WaitForPodDeletions(ctx context.Context, k8sClient client.Client, pods []co
 type WaitFunc = func() error
 
 func waitParallel(waiters []WaitFunc) error {
+	var errs error
 	errChan := make(chan error)
 	for _, waiter := range waiters {
 		go func(waiter func() error) {
 			errChan <- waiter()
 		}(waiter)
 	}
-	combinedMessage := ""
 	count := 0
 	for err := range errChan {
-		if err != nil {
-			combinedMessage += fmt.Sprintf("%s + ", err)
-		}
+		errs = errors.Join(errs, err)
```

Comment:
errors.Join is problematic for daemons because it separates errors with newlines, thus breaking errors into several separate log events with our current system.